config/datastores/posixDatastore.yaml

```diff
@@ -8,7 +8,7 @@ datastore:
     # valid_first and valid_last here are YYYYMMDD; we assume we'll switch to
     # MJD (DM-15890) before we need more than day resolution, since that's all
     # Gen2 has.
-    default: "{collection}/{datasetType}.{component:?}/{skymap:?}/{tract:?}/{patch:?}/{label:?}/{abstract_filter:?}/{instrument:?}/{physical_filter:?}/{calibration_label:?}/{visit:?}/{exposure:?}/{detector:?}"
+    default: "{collection}/{datasetType}.{component:?}/{tract:?}/{patch:?}/{label:?}/{abstract_filter:?}/{physical_filter:?}/{visit:?}/{instrument:?}{run}_{skymap:?}_{datasetType}_{component:?}_{tract:?}_{patch:?}_{label:?}_{abstract_filter:?}_{physical_filter:?}_{calibration_label:?}_{visit:?}_{exposure:?}_{detector:?}"
```

Comment:
Do you want instrument and run to run into each other?

Reply:
I actually did it on purpose but not for any good/real reasons (run is always a number).  Happy to have underscore there too.

Reply:
Some datatypes don't have `instrument` set; we don't mind having some filenames starting with `_` right?

Reply:
Good point.  Maybe leave the underscore out then.  Not entirely sure what the correct fix is. Possibly remove strings between optional and mandatory if there are no `[a-zA-Z]` characters...

Reply:
Another take...  Filenames are now led by `datasetType`; probably better aesthetically speaking although it's only a personal preference.